app/services/katello/pulp/repository.rb

```diff
@@ -139,7 +140,8 @@ def needs_importer_updates?
         repo_details = backend_data
         return unless repo_details
         capsule_importer = repo_details["importers"][0]
-        !importer_matches?(capsule_importer)
+        generated_importer = generate_importer
+        !importer_matches?(generated_importer, capsule_importer)
```

Comment:
another nitpick, but i'd probably not store the generate_importer return output and just call it in the next line